### PR TITLE
[test] Extract 45 more tests from SolidityEndToEndTest.cpp

### PR DESCRIPTION
**Pull request by aarlt**

### CODE REVIEW

**Comment by aarlt** [comment]
@chriseth I'm not sure what are the reasons for these errors. I'm not able to reproduce them. Can you take a look?

File: test/libsolidity/semanticTests/extracted/delegatecall_return_value.sol [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    function assert0() external view {
+        assert(value == 0);
+    }
+
+    function assert0_delegated() external returns (bool, bytes memory) {
+        return address(this).delegatecall(abi.encodeWithSignature("assert0()"));
+    }
+}
+// ----
```
Comment (chriseth) [inline test/libsolidity/semanticTests/extracted/delegatecall_return_value.sol]: needs byzantium

File: test/libsolidity/semanticTests/extracted/staticcall_for_view_and_pure.sol [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    function fview() public returns (uint256) {
+        return (CView(address(new C()))).f();
+    }
+
+    function fpure() public returns (uint256) {
+        return (CPure(address(new C()))).f();
+    }
+}
+// ----
```
Comment (chriseth) [inline test/libsolidity/semanticTests/extracted/staticcall_for_view_and_pure.sol]: needs to be split in different tests for post- and pre-byzantium

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 3]
```diff
@@ -6840,110 +6014,6 @@ BOOST_AUTO_TEST_CASE(bare_call_return_data)
...
-		ABI_CHECK(callContractFunction("assert0_delegated()"), encodeArgs(u256(0), 0x40, 0x00));
-		ABI_CHECK(callContractFunction("get_delegated()"), encodeArgs(u256(1), 0x40, 0x20, 1));
-		ABI_CHECK(callContractFunction("set(uint256)", u256(42)), encodeArgs());
-		ABI_CHECK(callContractFunction("get()"), encodeArgs(u256(42)));
-		ABI_CHECK(callContractFunction("assert0_delegated()"), encodeArgs(u256(0), 0x40, 0x00));
-		ABI_CHECK(callContractFunction("get_delegated()"), encodeArgs(u256(1), 0x40, 0x20, 42));
-	}
-	else
-	{
-		char const* sourceCode = R"DELIMITER(
```
Comment (chriseth) [inline test/libsolidity/SolidityEndToEndTest.cpp:6882]: This part is missing - will add a test myself.

File: test/libsolidity/semanticTests/extracted/staticcall_for_view_and_pure_pre_byzantium.sol [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    function fpure() public returns (uint256) {
+        return (CPure(address(new C()))).f();
+    }
+}
+// ====
+// EVMVersion: <byzantium
+// ----
+// f() -> 0x1 # This should work, next should throw #
```
Comment (chriseth) [inline test/libsolidity/semanticTests/extracted/staticcall_for_view_and_pure_pre_byzantium.sol]: Need to change comment.

**Review by chriseth — APPROVED** [review]

**Pull request merged**